homeassistant/components/synology_dsm/media_source.py

```diff
@@ -231,16 +250,19 @@ async def get(
         if not self.hass.data.get(DOMAIN):
             raise web.HTTPNotFound
         # location: {cache_key}/{filename}
-        cache_key, file_name = location.split("/")
+        cache_key, file_name, passphrase = location.split("/")
         image_id = int(cache_key.split("_")[0])
+
         if shared := file_name.endswith(SHARED_SUFFIX):
             file_name = file_name.removesuffix(SHARED_SUFFIX)
+
         mime_type, _ = mimetypes.guess_type(file_name)
         if not isinstance(mime_type, str):
             raise web.HTTPNotFound
+
         diskstation: SynologyDSMData = self.hass.data[DOMAIN][source_dir_id]
         assert diskstation.api.photos is not None
-        item = SynoPhotosItem(image_id, "", "", "", cache_key, "", shared)
+        item = SynoPhotosItem(image_id, "", "", "", cache_key, "", shared, passphrase)
         try:
             image = await diskstation.api.photos.download_item(item)
```

Comment:
```suggestion
        item = SynoPhotosItem(image_id, "", "", "", cache_key, "xl", shared, passphrase)
        try:
            image = await diskstation.api.photos.download_item_thumbnail(item)
```

`download_item` is not supported for shared items (throws 804 error). 
`download_item_thumbnail` with `xl` size works for every type of item, and is what the Synology Photos web app uses. 
Eventually, we can still use `download_item` for items without passphrase, to keep getting original files, not sure if that's worth it.

Reply:
Hi @alexismarquis thanks for testing and providing this feedback :+1: i've updated this PR, further tests would be highly appreciated